src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java

```diff
@@ -46,13 +55,36 @@ public final void setConnectionValues(BitBucket api, ConnectionValues values) {
 
     @Override
     public final UserProfile fetchUserProfile(BitBucket api) {
-        BitBucketUser user = api.userOperations().getUserWithRepositories()
-                .getUser();
+        BitBucketUser user = api.userOperations().getUserWithRepositories().getUser();
+        String primaryEmail = fetchPrimaryEmailIfAvailableForUserProfile(api, user);
         return new UserProfileBuilder().setFirstName(user.getFirstName())
-                .setLastName(user.getLastName())
+                .setLastName(user.getLastName()).setEmail(primaryEmail)
                 .setUsername(user.getUsername()).build();
     }
 
+    private String fetchPrimaryEmailIfAvailableForUserProfile(BitBucket api, BitBucketUser user) {
+        try {
+            List<BitBucketEmailAddress> emailAddresses = api.usersOperations()
+                    .emailsOperations().getListOfUserEmailAddresses(user.getUsername());
+            for (BitBucketEmailAddress emailAddress : emailAddresses) {
+                if (emailAddress.getPrimary()) {
+                    return emailAddress.getEmail();
+                }
+            }
+        } catch (HttpClientErrorException clientError) {
+            if (!HttpStatus.FORBIDDEN.equals(clientError.getStatusCode())) {
+                throw clientError;
+            } else {
+                if (logger.isWarnEnabled()) {
+                    logger.warn("BitBucket consumer configuration: Permissions:Acount:Email is not granted."
```

Comment:
I was going to suggest to add a comment explaining your intention, but the log message makes that clear. :+1: